AUV/auv_viewer_processing/auv_processing.py

```diff
@@ -0,0 +1,725 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*
+"""
+TODO
+-improve reporting function
+"""
+
+from datetime import datetime, timedelta
+from functools import partial
+from geopy.distance import vincenty
+from multiprocessing import Pool, cpu_count
+from netCDF4 import Dataset, num2date
+from osgeo import osr, gdal
+import StringIO
+import argparse
+import csv
+import operator
+import os
+import re
+import shutil
+import subprocess
+import sys
+import urllib2
+import uuid
+sys.path.insert(0, os.path.join(os.environ.get('DATA_SERVICES_DIR'), 'lib'))
+from python.imos_logging import IMOSLogging
+
+AUV_WIP_DIR = os.path.join(os.environ.get('WIP_DIR'), 'AUV', 'AUV_VIEWER_PROCESSING')
+
+def list_geotiff_dive(dive_path):
+    """
+    list the geotiffs images of one dive. Only looking for the left geotiff
+    """
+    geotiff_dir_dive_path = _geotiff_dive_path(dive_path)
+    geotiff_list          = []
+    for file in os.listdir(geotiff_dir_dive_path):
+        if file.endswith("LC16.tif"):
+            geotiff_list.append(os.path.join(geotiff_dir_dive_path, file))
+
+    geotiff_list.sort()
+
+    return geotiff_list
+
+def geotiff_corner_coordinates(geotiff_path):
+    """
+    retrieves the latlon coordinate corners of a geotiff image file
+    """
+    ds     = gdal.Open(geotiff_path)
+    old_cs = osr.SpatialReference()
+    old_cs.ImportFromWkt(ds.GetProjectionRef())
+
+    # create the new coordinate system
+    wgs84_wkt = """
+    GEOGCS["WGS 84",
+        DATUM["WGS_1984",
+            SPHEROID["WGS 84",6378137,298.257223563,
+                AUTHORITY["EPSG","7030"]],
+            AUTHORITY["EPSG","6326"]],
+        PRIMEM["Greenwich",0,
+            AUTHORITY["EPSG","8901"]],
+        UNIT["degree",0.01745329251994328,
+            AUTHORITY["EPSG","9122"]],
+        AUTHORITY["EPSG","4326"]]"""
```

Comment:
Examples [here](https://pcjericks.github.io/py-gdalogr-cookbook/projection.html) suggest to me you should be able to import the coordinate system just using the EPSG code.